### PR TITLE
Adds a syndicate AI card for nuke ops

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Adds a syndicate AI card for nuke ops. Costs 12 TC, can be refunded. Activating it in-hand opens up a ghost poll like normal reinforcements.
## Why It's Good For The Game

> Adds a syndicate AI card for nuke ops. Costs 12 TC, can be refunded. Activating it in-hand opens up a ghost poll.

It'll be fun to have a lil modsuit pal with you who can maybe help if you get stunned, in critical condition, or else. With an injector module they can even save your life, if it's loaded with medicine! Unless they can't actually inject, but oh well, out of scope.

12 TC seems fair to me, not so expensive that nobody's going to bother with something that's not really useful in the vast majority of circumstances, but not so cheap that everyone will get one.

There are also doubtless a huge amount of gimmicks that savvy nukies can get up to, having a pocket AI.

**Since this needs two players to be fully tested, I wasn't able to do so. Please testmerge this before merging so we can confirm it works ingame!**
## Changelog
:cl:
add: Adds a syndicate AI card for nuke ops. Costs 12 TC, can be refunded. Activating it in-hand opens up a ghost poll like normal reinforcements. Base interaction range for syndicate AIs is one, which means they can handle electronics only in proximity.
add: Adds a syndicate AI interaction range upgrade for nuke ops, costs 4 TC and can be applied onto a syndicate AI (inside any container) to increase its interaction range by two per. (Three purchases are recommended for seven tiles of range!)
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
> Since this needs two players to be fully tested, I wasn't able to do so

You can test content with two players locally by logging out of byond and joining your server as a guest.

**Comment by Jackal-boop** [comment]
this would give an operative all access to the station with the ability to bolt and electrify doors for 12 TC

**Comment by carlarctg** [comment]
> this would give an operative all access to the station with the ability to bolt and electrify doors for 12 TC. ususally you have to spend about 50TC on a borg for that

What? Huh??? This is an intelliCard, not an AI core. To my knowledge, those can't control the station's electronics. And AI cores need a board to be built.

**Comment by EricZilla** [comment]
They can control everything they can see

**Comment by carlarctg** [comment]
Well that's fucking stupid

**Comment by carlarctg** [comment]
I'll look into disabling that feature on syndicards. I might add an optional bonus item that unlocks that feature, that sounds fun.

**Comment by ATH1909** [comment]
you could just

use a syndiborg

**Comment by Vincent983** [comment]
there's a normal intellicard by default in the syndicate infirltrator, so if a nukie wanted the wireless features back, they could just put AI into modsuit, then upload AI to normal intellicard

**Comment by carlarctg** [comment]
> there's a normal intellicard by default in the syndicate infirltrator, so if a nukie wanted the wireless features back, they could just put AI into modsuit, then upload AI to normal intellicard

SIGH

**Comment by Serxule** [comment]
ok gotta be honest, this could lead to them making a syndicate AI, or a modsuit AI, both of which would be very, VERY strong, the syndi AI is just completely untouchable and would force crew to spend time cutting cameras, and the modsuit AI would 1. speed up movement as a whole, 2. bypass damage slowdown to an extent, 3. move a downed operative to safety

**Comment by Vincent983** [comment]
Building a full AI core for an syndicate AI would take the same effort as just turning a fellow operative into an AI. You don't even need lawboards, because nuclear operatives can buy a syndicate surgical bag which includes a syndicate MMI which slaves the built AI to you.

Also, you need an AI core board, which is only obtainable from the RD office or after research in order to have a full fledged AI core.

**Comment by EricZilla** [comment]
I think the best way to make this work would have to be a buyable syndicate suit that has a preinstalled AI, that is partially gimped

**Comment by carlarctg** [comment]
> ok gotta be honest, this could lead to them making a syndicate AI, or a modsuit AI, both of which would be very, VERY strong, the syndi AI is just completely untouchable and would force crew to spend time cutting cameras, and the modsuit AI would 1. speed up movement as a whole, 2. bypass damage slowdown to an extent, 3. move a downed operative to safety

Yes that's, uh. The whole point of the item...?

If a modsuit AI is somehow 'op' the tc cost can be increased. But I have a hard time believing that when it needs cooperation between 2 people on a level higher than the Savannah-Ivanov.

**Comment by MrDoomBringer** [comment]
Could always just up the TC cost to like 50

**Comment by Vincent983** [comment]
a syndicate AI that is stuck in a modsuit definitely isn't worth nearly as much as an assault borg

**Comment by Loafin34** [comment]
could this also disable microbomb implants in the modsuit AI? i feel like that'd be good if you as an AI are trying to save your nukie bud who got speared too many times

(or at least make it so it doesnt activate on succumb)

File: code/game/objects/items/devices/aicard.dm [hunk 1]
```diff
@@ -13,6 +13,80 @@
...
+	var/datum/antagonist/nukeop/creator_op = user.mind?.has_antag_datum(/datum/antagonist/nukeop,TRUE)
+	if(!creator_op)
+		return
+	var/list/nuke_candidates = poll_ghost_candidates("Do you want to play as a syndicate artifical intelligence inside an intelliCard?", ROLE_OPERATIVE, ROLE_OPERATIVE, 150, POLL_IGNORE_SYNDICATE)
+	if(LAZYLEN(nuke_candidates))
+		if(QDELETED(src))
+			return
+		used = TRUE
+		// pick ghost, create AI and transfer
+		var/mob/dead/observer/ghos = pick(nuke_candidates)
```
Comment (Ghommie) [inline code/game/objects/items/devices/aicard.dm:72]: have you dropped a t on purpose?

File: code/game/objects/items/devices/aicard.dm [hunk 2]
```diff
@@ -13,6 +13,80 @@
...
+	if(AI || used)
+		return ..()
+	procure_ai(user)
+
+/obj/item/aicard/syndie/loaded/proc/procure_ai(mob/user)
+	var/datum/antagonist/nukeop/creator_op = user.mind?.has_antag_datum(/datum/antagonist/nukeop,TRUE)
+	if(!creator_op)
+		return
+	var/list/nuke_candidates = poll_ghost_candidates("Do you want to play as a syndicate artifical intelligence inside an intelliCard?", ROLE_OPERATIVE, ROLE_OPERATIVE, 150, POLL_IGNORE_SYNDICATE)
+	if(LAZYLEN(nuke_candidates))
```
Comment (Ghommie) [inline code/game/objects/items/devices/aicard.dm:67]: Early return if there's no candidate instead.

File: code/game/objects/items/devices/aicard.dm [hunk 3]
```diff
@@ -13,6 +13,80 @@
...
+	max_capacity = 1000
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/computer_disk/syndie_ai_upgrade/pre_attack(atom/A, mob/living/user, params)
+	var/mob/living/silicon/ai/AI
+	if(isAI(A))
+		AI = A
+	else
+		AI = locate() in A
+	if(AI && AI.interaction_range != INFINITY)
```
Comment (Ghommie) [inline code/game/objects/items/devices/aicard.dm:30]: early returns.

**Review by Ghommie — CHANGES_REQUESTED** [review]
The only icons this is adding are a palette swap of the AI card item, I don't think this needs a `Needs Pictures` label. Anyhow, merge conflicts.

File: code/game/objects/items/devices/aicard.dm (hunk 1 above)
Comment (carlarctg) [reply code/game/objects/items/devices/aicard.dm:72]: yes

**Comment by ZephyrTFA** [comment]
> The only icons this is adding are a palette swap of the AI card item, I don't think this needs a `Needs Pictures` label. Anyhow, merge conflicts.

Read the description of the label.

**Comment by Ghommie** [comment]
Oh, my bad then!

**Comment by carlarctg** [comment]
I forgor to get an screenshot ill do it later

**Comment by carlarctg** [comment]
image

@ZephyrTFA @Ghommie here you goa re you HAPPY

**Comment by ZephyrTFA** [comment]
don't sass me, I don't need it I get enough from oranges

**Pull request merged**